pandas/core/indexes/period.py

```diff
@@ -1035,5 +1059,7 @@ def period_range(start=None, end=None, periods=None, freq='D', name=None):
         raise ValueError('Of the three parameters: start, end, and periods, '
                          'exactly two must be specified')
 
-    return PeriodIndex(start=start, end=end, periods=periods,
-                       freq=freq, name=name)
+    data, freq = PeriodArray._generate_range(start, end, periods, freq,
```

Comment:
Note, there's a slight bit of duplication here with the PeriodIndexConstructor.

We can't call `PeriodIndex.__init__` from here, since then we hit the warning.

We can't call `period_range` from `PeriodIndex.__init__` since `period_range` doesn't handle fields.

In the end, it's just two lines, so I think we're OK.